pmd-core/src/main/java/net/sourceforge/pmd/internal/Slf4jSimpleConfiguration.java

```diff
@@ -41,19 +44,42 @@ public static void reconfigureDefaultLogLevel(Level level) {
             initMethod.setAccessible(true);
             initMethod.invoke(null);
 
-            // Call SimpleLoggerFactory.reset() by reflection.
-            Method resetMethod = loggerFactoryClass.getDeclaredMethod("reset");
-            resetMethod.setAccessible(true);
-            resetMethod.invoke(loggerFactory);
+            int newLogLevel = getDefaultLogLevelInt(simpleLoggerClass);
+
+            Field currentLogLevelField = simpleLoggerClass.getDeclaredField("currentLogLevel");
+            currentLogLevelField.setAccessible(true);
+
+            // Change the logging level of loggers that were already created.
+            // For this we fetch the map of name to logger that is stored in the logger factory,
+            // then set the log level field of each logger via reflection.
+            Field loggerMapField = loggerFactoryClass.getDeclaredField("loggerMap");
+            loggerMapField.setAccessible(true);
+            @SuppressWarnings("unchecked")
+            Map<String, Logger> loggerMap = (Map<String, Logger>) loggerMapField.get(loggerFactory);
+            for (Logger logger : loggerMap.values()) {
+                if (logger.getName().startsWith(PMD_ROOT_LOGGER)
+                    && SIMPLE_LOGGER_CLASS.equals(logger.getClass().getName())) {
+                    currentLogLevelField.set(logger, newLogLevel);
+                }
+            }
         } catch (ReflectiveOperationException ex) {
             System.err.println("Error while initializing logging: " + ex);
         }
 
         PmdLoggerFactoryFriend.reset();
     }
 
+    private static int getDefaultLogLevelInt(Class<?> simpleLoggerClass) throws ReflectiveOperationException {
+        Field configParamsField = simpleLoggerClass.getDeclaredField("CONFIG_PARAMS");
+        configParamsField.setAccessible(true);
+        Object configParams = configParamsField.get(null);
+        Field defaultLogLevelField = configParams.getClass().getDeclaredField("defaultLogLevel");
+        defaultLogLevelField.setAccessible(true);
+        return (int) defaultLogLevelField.get(configParams);
+    }
+
     public static Level getDefaultLogLevel() {
-        Logger rootLogger = LoggerFactory.getLogger(Logger.ROOT_LOGGER_NAME);
+        Logger rootLogger = LoggerFactory.getLogger(PMD_ROOT_LOGGER);
```

Comment:
It probably doesn't make a big difference in practice (only if someone indeed confiugres the logging via simplelogger.properties), but only the root logger would have the default log level. The pmd root logger (`net.sourceforge.pmd`) could have a different log level configured.
So, I'd leave it as before.